### ISSUE TITLE
Troubleshooting Unexpected Server Crashes

### ISSUE THREAD

**Issue by MihailHavenT** [SmartlyDressedGames/Unturned-3.x-Community#4762]
Hello everyone,

I'm here to understand why some of my servers are shutting down unexpectedly. There’s a crash occurring that I can’t quite identify, and I'm unsure what’s causing it. Is there anyone with expertise in this area who has experienced a similar issue and managed to resolve it? The console shows a 'Crash!' message, followed by a long text.

I look forward to any responses or suggestions.

**Comment by asineth0** [comment]
Does the crash have anything to do with `steamclient.so`?

**Comment by sunnamed434** [comment]
Can you send the crash message?

**Comment by MihailHavenT** [comment]
> Does the crash have anything to do with `steamclient.so`?
I have disabled the Fake_IP option in the server configuration, and now I see that there are no more crashes! I contacted a server manager for Unturned, and he tell me to disable this option because there's an issue with Steam’s servers. Apparently, Nelson (the game's developer) is aware of the problem and is actively seeking a solution. He's also in contact with Steam's support team to address this issue. (That's what said that server manager.)

> Can you send the crash message?
I saw something on the console, but I didn't have time to copy the contents. The first message I noticed was a simple text that said 'Crash!' followed by a long explanation about the crash.

**Comment by asineth0** [comment]
So I did some testing and I'm pretty certain it's FakeIP. If you get unlucky and Steam servers drop out for a second, you get a crash (but only sometimes). Going to have to disable it on my servers for now.

**Comment by MihailHavenT** [comment]
I hope Nelson will see this issue ticket on GitHub to find out why it would cause this... @SDGNelson I'm waiting for your answer!

**Comment by DanielWillett** [comment]
He's just going to ask for logs, U3DS/Logs/Server_[ServerName].log

**Comment by MihailHavenT** [comment]
I put the server logs for nothing; there is nothing there...

**Comment by SDGNelson** [comment]
If it is the Fake IP crash, then we're currently waiting to see if we get a response on the GNS repo.

Is it the same issue you submitted a support ticket about the other day though?

**Comment by MihailHavenT** [comment]
Yes, I don’t see any error related to the crash in the logs, and I currently have no way to send them to you. I need to stay alert when the server shuts down to copy everything displayed in the console. At the moment, all my servers are not using fake IPs!

**Comment by asineth0** [comment]
> Yes, I don’t see any error related to the crash in the logs, and I currently have no way to send them to you. I need to stay alert when the server shuts down to copy everything displayed in the console. At the moment, all my servers are not using fake IPs!

In my experience, the FakeIP crash doesn't show up in logs, only in console. I think this would apply for any crash in native code.

**Comment by MihailHavenT** [comment]
![image](https://github.com/user-attachments/assets/1e43e2ee-4574-43b4-b64a-682ccd419256)
[Crash!!!.txt](https://github.com/user-attachments/files/18262381/Crash.txt)
It crashes constantly on some servers, almost every 5 hours consecutively!
